test.js

```diff
@@ -44,3 +44,34 @@ test.failing('dynamic import works', async t => {
 		})(),
 	);
 });
+
+test('iterator object', async t => {
+	const fixture = [1, 2];
+
+	const asyncIterator = await makeAsynchronousIterator(fixture => fixture[Symbol.iterator]())(fixture);
+	const result = [];
+
+	for await (const value of asyncIterator) {
+		result.push(value);
+	}
+
+	t.deepEqual(result, fixture);
+});
+
+test('generator function', async t => {
+	const fixture = [1, 2];
+
+	const asyncIterator = await makeAsynchronousIterator(function * (fixture) {
+		for (const value of fixture) {
+			yield value;
+		}
+	})(fixture);
+
+	const result = [];
+
+	for await (const value of asyncIterator) {
+		result.push(value);
+	}
+
+	t.deepEqual(result, fixture);
+});
```

Comment:
I'd like to see a test where it throws in the middle of iterating an async iterator. Just to make sure error handling is handled properly.